src/components/modals/ModalUserEdit.tsx

```diff
@@ -0,0 +1,363 @@
+import { Box, Button, Grid, Modal, SimpleGrid, Space, Text, TextInput, Title } from "@mantine/core";
+import React, { useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import SortableTree from "react-sortable-tree";
+import FileExplorerTheme from "react-sortable-tree-theme-file-explorer";
+import { Mail, User } from "tabler-icons-react";
+
+import { signup } from "../../actions/authActions";
+import { fetchDirectoryTree, manageUpdateUser, updateUserAndScan } from "../../actions/utilActions";
+import { useAppDispatch, useAppSelector } from "../../store/store";
+import { PasswordEntry } from "../settings/PasswordEntry";
+
+type Props = {
+  isOpen: boolean;
+  updateAndScan?: boolean;
+  userToEdit: any;
+  selectedNodeId?: string;
+  onRequestClose: () => void;
+  userList: any;
+  createNew: boolean;
+  firstTimeSetup?: boolean;
+};
+
+export function ModalUserEdit(props: Props) {
+  const { isOpen, updateAndScan, selectedNodeId, onRequestClose, userList, createNew, firstTimeSetup } = props;
+  var userToEdit = props.userToEdit;
+  const [newScanDirectory, setNewScanDirectory] = useState("");
+  const [treeData, setTreeData] = useState([]);
+  const [userName, setUserName] = useState("");
+  const [userNameError, setUserNameError] = useState("");
+  const [userEmail, setUserEmail] = useState("");
+  const [userEmailError, setUserEmailError] = useState("");
+  const [userFirst, setUserFirst] = useState("");
+  const [userLast, setUserLast] = useState("");
+  const [userPassword, setUserPassword] = useState("");
+  const [newPasswordIsValid, setNewPasswordIsValid] = useState(true);
+
+  const [isAdmin, setIsAdmin] = useState(false);
+  const [scanDirectoryPlaceholder, setScanDirectoryPlaceholder] = useState("");
+  const dispatch = useAppDispatch();
+  const auth = useAppSelector(state => state.auth);
+  const { directoryTree } = useAppSelector(state => state.util);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const { t } = useTranslation();
+  const [pathDoesNotExist, setPathDoesNotExist] = useState(false);
+  const [pathError, setPathError] = useState("");
+  const [closing, setClosing] = useState(false);
```

Comment:
All the separate usages of `useState` will cause a re-render every time it changes. I think there's two options here.

1. Split up the component into smaller sub-components (one from each input field). 
2. Use the `useReducer` function here.

Personally, I'm in favor of `1` because it breaks this component up into smaller pieces, which are easier to digest when debugging. On the other hand, we'd have a hand-full of small components lying around which only get used once.

Reply:
I am also in favor of splitting this up, but a lot of this is code from the legacy component. So this would be an optional fix from my point of view.